patches/net/minecraft/client/renderer/entity/state/EntityRenderState.java.patch

```diff
@@ -1,10 +1,39 @@
 --- a/net/minecraft/client/renderer/entity/state/EntityRenderState.java
 +++ b/net/minecraft/client/renderer/entity/state/EntityRenderState.java
-@@ -27,6 +_,7 @@
+@@ -7,7 +_,7 @@
+ import net.neoforged.api.distmarker.OnlyIn;
+ 
+ @OnlyIn(Dist.CLIENT)
+-public class EntityRenderState {
++public class EntityRenderState extends net.neoforged.neoforge.attachment.AttachmentHolder implements net.neoforged.neoforge.client.extensions.IRenderStateExtension {
+     public double x;
+     public double y;
+     public double z;
+@@ -27,6 +_,27 @@
      public Vec3 nameTagAttachment;
      @Nullable
      public EntityRenderState.LeashState leashState;
 +    public float partialTick;
++    private java.util.Map<net.minecraft.util.context.ContextKey<?>, Object> extensions = java.util.Map.of(); // NeoForge: allow generic extension of render states
```

Comment:
I can't think of any cases where it would be worth clearing tbh. Do you have an example of when it would be cleared?

Reply:
It would be cleared every time the render state extraction happens, right before the modifiers are called. This would prevent state leaks between entities when the extension is conditional as I would expect people to just do something like this:
```java
Thing thing = computeData(entity);
if (thing != null)
    state.setRenderData(KEY, thing);
```
instead of always writing the result to the extension.

Reply:
ah, rather than just passing null and removing the object. Sure, I suppose for safety reasons it makes sense to do that.